### DISCUSSION TITLE
Status of the storefront app is very unclear

### DISCUSSION THREAD

**Discussion by olivierdalang** [saleor/saleor#8383] [Q&A] [answered]
Hi !

It's very unclear whether the storefront app is considered as part of the stack and can be used as a starting point to create a online shop, or whether users are expected to develop their own frontend more or less from scratch, which obviously represents a large effort.

From the website (which includes UI/UX in the selling points and shows the frontend as a demo), I tend to think that it is part of the stack and that it works more or less perfectly out of the box.

From the storefront repo, I tend to think it's merely an example implementation to take inspiration from, but that it's not ready to be used, mainly because of this note:

> Note: This project is a demonstration on how Saleor can be used. It’s not ready to be a starter but rather show how different cases can be handled and could be used as a recipe book. There will be breaking changes and the code is constantly evolving, so use at your own risk.

So what exactly is the status of it ?

- Is that message just there because there's no extension/theming mechanism, meaning that customizing it also means loosing ability to painlessly upgrade ? (which would anyway be the case with a from scratch solution)

- Is the store-front feature complete regarding advertised saleor's features ?

- Is it production ready ?

Thanks !

**Comment by PlusA2M** [comment]
Hi! I'm new to this community also, let's start with my perspective on this topic.
It's quite clear what that note message tell, it is **not** a starter but you could use and customise on your own. 
Saleor is built with headless architecture hence it is nice to use it with your tailor-made frontend (that's why we have the wonderful GraphQL API, isn't it?). 
The `saleor-storefront` is one of the project that team Saleor want to show us how it would be like on the frontend side, even set the standard on speed, architecture, etc. for other developers who want to make use of `saleor-core`. However, I think it's **not** the main focus for the team so it would be production ready soon but not on the first priority. I can see they have put the rebase of `saleor-storefront` on their roadmap, you can also check it out from [here](https://github.com/saleor/saleor/projects/12).

**Comment by patrys** [comment, marked as answer]
Here's where the storefront stands currently. We created the `saleor-storefront` repository to show how Saleor can be used to build an actual store. It's the same repository that we're using to run `demo.saleor.io`.

Unfortunately, it turns out it's too close to an actual store, and people often judge it by its (outdated) looks. So we think a fully transactional wireframe would work better.

It's also old enough to have a lot of tech debt, making it hard to maintain and customize. That's why we can't honestly recommend it as a starter kit.

Because of the above reasons, we've decided to build a new storefront starting from scratch, with no fancy design, focusing solely on best practices for building commerce storefronts. The new project, called `react-storefront`, uses Next.js as its foundation and Tailwind for the looks. You can find it here: https://github.com/saleor/react-storefront. Over time, it will replace the current `saleor-storefront` project, but it's still missing some features like account management for logged-in users.

You can see the latest build in action by visiting https://learn-saleor-demo.vercel.app.

**Reply by olivierdalang** [reply]
Awesome, thanks for this very clear answer.

I was just gonna suggest to add a reference to that in the storefront's repo but saw it was done already 👍

**Reply by juanzgc** [reply]
It seems https://learn-saleor-demo.vercel.app is no longer active - Do you mind please updating the link

**Reply by zaiste** [reply]
@juanzgc We changed the URL to https://reactstorefront.vercel.app Apologies for the inconvenience.
